### DISCUSSION TITLE
Create a migration schema with a custom type that serializes to JSON

### DISCUSSION THREAD

**Discussion by Kayanski** [SeaQL/sea-orm#2019] [Q&A]
Hello,
I'm trying to create a table with some custom types inside of it. I don't want to store those types as is but I just want them to be automatically serialized and deserialized on insert/update/select...

I see that support for it exists at the entity level : see [the end of this paragraph](https://www.sea-ql.org/SeaORM/docs/generate-entity/entity-structure/#column-type).

However, I would prefer my entities to be auto-generated from Migration Schemas, and I see no support for this feature inside the `ColumnDef` object for instance.
Does it exist already in the library ?

My solution today is to create a `json` column and after entity generation, I manually change the type of the column, which is not ideal obviously?

An example would be able to generate the following from the CLI : 
```rust
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name = "tx")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub events: TxLogs,
}

/// In another file
/// This is the structure I want to be created when loading the Model
#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, FromJsonQueryResult)]
pub struct TxLogs(String);

```

Thank you very much

**Comment by lucky9-cyou** [comment]
I have also encountered a similar problem. May I ask if there is a better solution now?

**Reply by askasp** [reply]
Now i'm the same place as you guys. No luck finding a better solution in the last 9 months?
